Source/Scenes/World.cs

```diff
@@ -6,288 +8,310 @@ namespace Celeste64;
 
 public class World : Scene
 {
-	public enum EntryReasons { Entered, Returned, Respawned }
-	public readonly record struct EntryInfo(string Map, string CheckPoint, bool Submap, EntryReasons Reason);
-
-	public Camera Camera = new();
-	public Rng Rng = new(0);
-	public float HitStun = 0;
-	public bool Paused = false;
-	public EntryInfo Entry = new();
-	public readonly GridPartition<Solid> SolidGrid = new(200, 100);
-	public float GeneralTimer = 0;
-	public float DeathPlane = -100;
-
-	public readonly List<Actor> Actors = [];
-	private readonly List<Actor> adding = [];
-	private readonly List<Actor> destroying = [];
-	public readonly Dictionary<Type, List<Actor>> tracked = []; // TAS: publicized
-	private readonly Dictionary<Type, Queue<Actor>> recycled = [];
-	public readonly List<Type> trackedTypes = []; // TAS: publicized
-
-	private readonly List<ModelEntry> models = [];
-	private readonly List<Sprite> sprites = [];
-
-	private Target? postTarget;
-	private readonly Material postMaterial = new();
-	private readonly Batcher batch = new();
-	private readonly List<Skybox> skyboxes = [];
-	private readonly SpriteRenderer spriteRenderer = new();
-
-	// Pause Menu, only drawn when actually paused
-	private readonly Menu pauseMenu = new();
-	private AudioHandle pauseSnapshot;
+    public enum EntryReasons { Entered, Returned, Respawned }
+    public readonly record struct EntryInfo(string Map, string CheckPoint, bool Submap, EntryReasons Reason);
+
+    public Camera Camera = new();
+    public Rng Rng = new(0);
+    public float HitStun = 0;
+    public bool Paused = false;
+    public EntryInfo Entry = new();
+    public readonly GridPartition<Solid> SolidGrid = new(200, 100);
+    public float GeneralTimer = 0;
+    public float DeathPlane = -100;
+
+    public readonly List<Actor> Actors = [];
+    private readonly List<Actor> adding = [];
+    private readonly List<Actor> destroying = [];
+    public readonly Dictionary<Type, List<Actor>> tracked = []; // TAS: publicized
+    private readonly Dictionary<Type, Queue<Actor>> recycled = [];
+    public readonly List<Type> trackedTypes = []; // TAS: publicized
+
+    private readonly List<ModelEntry> models = [];
+    private readonly List<Sprite> sprites = [];
+
+    private Target? postTarget;
+    private readonly Material postMaterial = new();
+    private readonly Batcher batch = new();
+    private readonly List<Skybox> skyboxes = [];
+    private readonly SpriteRenderer spriteRenderer = new();
+
+    // Pause Menu, only drawn when actually paused
+    private readonly Menu pauseMenu = new();
+    private AudioHandle pauseSnapshot;
+
+    private static TASSettingsMenu? tassettingsmenu;
 
     // Used for mouse offset
-    private Vec2 nextMousePosition;
-    private Vec2 prevMousePosition;
+    public Vec2 nextMousePosition; //CelesteTAS: publicised
+    public Vec2 prevMousePosition; //CelesteTAS: publicised
```

Comment:
These fields were added by us. Tho it might make more sense to move them outside the `World`.